test/e2e/pkg/client/client.go

```diff
@@ -13,30 +13,140 @@
 package client
 
 import (
+	"fmt"
+	"io/ioutil"
+	"strconv"
+	"time"
+
+	"github.com/devfile/api/pkg/apis/workspaces/v1alpha2"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"log"
+	"os/exec"
+
 	"k8s.io/client-go/kubernetes"
-	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+var (
+	Scheme             = runtime.NewScheme()
+	SchemeBuilder      = runtime.NewSchemeBuilder(addKnownTypes)
+	AddToScheme        = SchemeBuilder.AddToScheme
+	SchemeGroupVersion = schema.GroupVersion{Group: v1alpha2.SchemeGroupVersion.Group, Version: v1alpha2.SchemeGroupVersion.Version}
+)
+
+func init() {
+	if err := AddToScheme(scheme.Scheme); err != nil {
+		log.Fatalf("Failed to add CRD to scheme")
+	}
+	if err := api.AddToScheme(Scheme); err != nil {
+		log.Fatalf("Failed to add CRD to scheme")
+	}
+}
+
 type K8sClient struct {
-	kubeClient *kubernetes.Clientset
+	kubeClient  *kubernetes.Clientset
+	crClient    crclient.Client
+	kubeCfgFile string // generate when client is created and store config there
 }
 
-// NewK8sClient creates kubernetes client wrapper with helper functions and direct access to k8s go client
-func NewK8sClient() (*K8sClient, error) {
-	cfg, err := config.GetConfig()
+// NewK8sClientWithKubeConfig creates kubernetes client wrapper with the specified kubeconfig file
+func NewK8sClientWithKubeConfig(kubeconfigFile string) (*K8sClient, error) {
+	cfg, err := clientcmd.BuildConfigFromFlags("", kubeconfigFile)
 	if err != nil {
 		return nil, err
 	}
+
+	//TODO we need admins KUBECONFIG or credentials as arguments
+	//TODO copy files with go
+	cfgBump := fmt.Sprintf("/tmp/admin.%s.kubeconfig", generateUniqPrefixForFile())
+	err = copyFile(kubeconfigFile, cfgBump)
+
+	if err != nil {
+		log.Fatal(fmt.Sprintf("Can't bump kubeconfig %s", err))
```

Comment:
I believe if possible we should return error but not fail with fatal.

Reply:
done